controllers/main.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+import time
+
+'''
+In this game we have two role - invader and hero. Both can bullet.
+
+Invaders are located at the top of game field and always move from the right side to the left and revert.
+Also they slowly move to the bottom.
+
+Main hero is located at the centre at the bottom. He can move to the left and to the right.
+If he bullet some invader, his bonus is grow.
+When invader bullet to main hero, the main hero's life is decrease.
+'''
+
+
+class Unit(object):
+
+    def __init__(self, unit_filename, bullet_filename, x, y, bonus=0, speed=5):
+        self.image = unit_filename
+        self.x = x
+        self.y = y
+        self.width = 10  # temporary when we don't have real images
+        self.height = 10  # must be height of real image
+        self.bonus = bonus
+        self.speed = speed
+        self.is_dead = False
+        self.bullet = Bullet(self, bullet_filename)
+        self.shift = 5
+
+    def move_to(self, x, y):
+        self.x = x
+        self.y = y
+
+    def check_fire(self, other_unit):
+        raise NotImplementedError
+
+    def fire(self, other_unit):
+        raise NotImplementedError
+
+    def make_bullet(self):
+        raise NotImplementedError
+
+
+class Invader(Unit):
+    moving_speed = 0
+
+    def __init__(self, hero_filename, bullet_filename, x, y, bonus=0, speed=5):
```

Comment:
In future commits I'd like to see a possibility of Invider or Unit instantiation without Image file specification (optional parameter). In that case we should be able to generate the image from random list. Initialization of each object must be as easy as possible, need to keep with less arguments